### ISSUE TITLE
Is this project still maintained?

### ISSUE THREAD

**Issue by ariporad** [shelljs/shelljs#272]
Hi,

Is this project still maintained? If not, would you be open to a new maintainer?

Thanks!

**Comment by nfischer** [comment]
I'd be open to help maintaining this project. I think the original maintainer intended to open it up to more maintainers.

I'm currently maintaining [my own fork](https://github.com/nfischer/shelljs) that has some of the not-yet-merged features. If you'd like to get some of the more recent features, feel free to use my fork (and/or submit upstream PRs there as well).

The idea is that hopefully the features that get added to that (and are therefore observed to be stable) will be more quickly merged upstream.

**Comment by ariporad** [comment]
@nfischer: I've been kind of talking with @arturadib [on Twitter](https://twitter.com/ariporad/status/685279595973574656). I'm hoping that he'll give someone (you, if you want, I also volunteer) push rights. He seems to be [inactive](https://twitter.com/ariporad/status/685279595973574656).

**Comment by nfischer** [comment]
Thanks for getting in contact with him. I'm not a twitter user, so I couldn't reach out to him. I'd personally really like to see this project be more actively maintained.

I'd definitely be interested in co-maintaining the project. I don't think I have enough experience with this particular project to know what features are most desired by the current users. I do however have decent experience in shell scripting (and can see areas where ShellJS lacks compared to Bash), and can review any POSIX-related pull requests (adding new commands, making ShellJS functions match POSIX behavior, etc.). I'd also be able to help go through old issues, verify bugs, and help provide simple solutions to problems users report. Having at least the permission to close issues would be very helpful there.

I think the best way to go forward would be to have @arturadib add one or two people to the ShellJS organization, and perhaps set up a channel for discussing major PRs and design decisions for the project. What do you think, @ariporad?

**Comment by ariporad** [comment]
@nfischer: well, I don't really have any experience, but I belive in this project, and would be happy to help our (if you'd like). I purposed you since you seemed active, but there might be others too.

Besides that, I think your plan sounds great! I think that whoever gets added to the org should just create a gitter, and we can chat there.

**Comment by nfischer** [comment]
I think a gitter is a great idea. Slack would be another alternative. I actually only recently found out about this project, but immediately found features/bug fixes, so I've been working on some PRs. When I saw that there were over 100 open issues, I went through and answered the ones I could, to help filter through stuff so the maintainer(s) would have an easier time later on.

I anticipate having a decent amount of spare time to devote to this, so I'd at the very least be able to respond to issues and bug fixes, if not also write my own feature additions as PRs.

**Comment by arturadib** [comment]
Added both of you as collaborators. Use it wisely :)

Main thing I ask is to (a) try to remain as close as possible to *nix
commands, (b) avoid feature bloat, and (c) only accept new
features/behaviors with tests that pass on CI.

Thanks guys!

@nfischer <https://github.com/nfischer>- if you create a Twitter account
I'll give a shout out to both of you on Twitter :)

-@r2r <http://twitter.com/r2r>

**Comment by nfischer** [comment]
@arturadib Thanks! Yeah, my thinking was to keep behavior close to *nix, and change things if current behavior doesn't match *nix. Also, do you know why the CI doesn't seem to be working right now? I created issue #266 for it. I imagine it has to do with renaming from `arturadib/shelljs` -> `shelljs/shelljs`, but I'm not completely sure.

@ariporad Would you be able to set up a gitter for the three of us to discuss the best way to go moving forward? I was thinking we should prioritize merging bug fix pull requests from other users, and then once those are merged, pull in bug fixes from each other, and then move on to feature additions afterward.

**Comment by arturadib** [comment]
CI should be fixed now, thanks for the heads up!
https://travis-ci.org/shelljs/shelljs

-@r2r <http://twitter.com/r2r>

**Comment by ariporad** [comment]
@arturadib: It looks like I need to be a member of the org (`shelljs`) to create the gitter. Any chance that would be possible?

Thanks!

**Comment by ariporad** [comment]
@arturadib, @nfischer: But while we're talking here, I have a few ideas:
1. I'm going to setup some better labels, is that OK?
2. I think we should make heavy use of those labels and of the assignment feature.
3. What are your thoughts about [`semantic-release`](http://npm.im/semantic-release)? I personally really like it.
4. Any objections to setting up AppVeyor (like travis for Windows)
5. Thoughts on [lgtm.co](http://ltgm.co) (I've never used it)?
6. I think it might be wise to switch to a traditional test runner, or at least add code coverage.
7. Contributing guidelines?

However, it seems like you'll be much more active a maintainer than I (based off the last 24h), so I'll defer to you to make decisions.

**Comment by nfischer** [comment]
1. Labels are great. Feel free to go ahead and label any old issues as well.
2. Again, using labels and assignments are fine.
3. I don't have any experience with that package (although I'm familiar with the concept of semver). My thoughts are that ShellJS isn't ready yet for v1.0, so most changes are up for grabs (without too much regard to the version number). Would that be useful even before we hit v1.0?
4. I've never used AppVeyor, but I'm not completely opposed. I think more of a priority is getting old PRs merged into the project, and later on adding in the extra CI. What do you think?
5. I like the idea of peer review. How difficult would it be to set up lgtm.co? I've never used that, but I've used a similar work flow before and liked it. I think passing the review of at least one maintainer should be good enough for approval (since we have quite a few PRs to work through).
6. The current unit tests are simple enough to write, and seem to do the job. Do you think there's a benefit to switching frameworks?
7. I think we should write up some contributing guidelines. We should probably have an in-depth discussion about what we want to be in the guidelines though. I personally like Vundle's policy of squashing commits (See [here](https://github.com/VundleVim/Vundle.vim/blob/master/CONTRIBUTING.md))

If you'd like, we can set up a slack team for the time being, and can set up a gitter later on.

**Comment by ariporad** [comment]
@arturadib: Sorry to keep bothering you, but it looks like we need admin perms for this repo to set stuff up, such as gitter and lgtm.io. (Actually, could we maybe get the ability to create repos in the org too, if possible?) Also, how should we contact you if we need to do a release (As of right now, we don't have npm perms, which is totally fine, but I'm not sure spamming your twitter is the best way).

@nfischer: 

Semantic Release would be helpful because it handles the releasing and the change log generation for us. It also ensures that anything on master is production ready (because it will be released to npm).

I think AppVeyor would be worthwhile because it allows us to reduce bugs on Windows before merging.

LGTM takes 5 minutes, as soon as I have the right perms, I'll set it up.

I think out current setup is fine, but I'm going to try to setup [`nyc`](http://npm.im/nyc), to get code coverage.

I'd rather not setup slack if possible, I already have a daunting number of slack logins (since there's no way to link accounts, grr...), but if needed, I'll set on up.

**Comment by ariporad** [comment]
@nfischer: Another thought, do you want to have us work on branches in this repo, or in our forks? (I don't think we should work on master).

**Comment by ariporad** [comment]
Also, Labels have been setup. They currently use the semantic-release/angular commit types, but that can be changed (they were created by a script).

**Comment by nfischer** [comment]
The new labels look great!

I think working on our own forks might be cleanest. That seems to work well for [Vimium](https://github.com/philc/vimium/pulls) at least.

You make good points about semantic release and appveyor. As long as setting those up doesn't impede us too much, it sounds good to me. I'm ok with moving forward without lgtm.co for a while, as long as we're careful. For the time being, let's not merge until each of us gives an LGTM on the PR.

**Comment by ariporad** [comment]
@nfischer: Ok, sounds great. Once we have perms, I'll set those up.

**Comment by ariporad** [comment]
@nfischer: Idea: Add a label for each command (ex. `shjs`, `ls`, `cd`, `touch`, `grep`).

**Comment by ariporad** [comment]
Also, are we going to do `semantic-release`? If so, we need to be _*really*_ strict about making sure commits conform to the [guidelines](https://github.com/ajoslin/conventional-changelog/blob/master/conventions/angular.md). We can also setup [GitCop](https://gitcop.com), which can enforce that for us.

**Comment by nfischer** [comment]
I think the command labels could be useful. Although it could be tedious to maintain that. I'm not opposed to the extra labels though (if they're too tedious, we can always just stop using them).

If semantic release is strict about commit messages, then this would be a pain to use for old PRs--it would basically mean we can't merge any of them unless the authors rewrite their commit messages and force-push to their forks. If that's the case, it's probably easier to wait on semantic-release and add it in later once we get some of these issues resolved. I don't think we're ready yet for another release, so it should be OK to wait on that. What do you think?

**Comment by ariporad** [comment]
@nfischer: It is, but the way it works is that all commits between the last release and the current release must match. But as far a PRs go, that's easy: https://git.io/vuFF1

**Comment by nfischer** [comment]
:+1: for the link.

Just might be tedious if we have to edit everyone's pull requests to get them to merge. Either way, it seems like we'll have to wait for perms to set up gitcop, semantic-release, etc. So until we get perms, I think we should stick with the current method of merging good PRs as-is. Since the previous commits don't follow semantic-release, this probably isn't making things much harder for when we switch.

**Comment by ariporad** [comment]
Sounds great. But shall we say use the commit guidelines going forward?

**Comment by nfischer** [comment]
Sounds good to me. I'll try to conform to conform to that in any commits I make.

**Comment by ariporad** [comment]
@nfischer: Thanks, no pressure.

ProTip: You're going to want to change the merge commit text when you merge commit messages. (I personally recommend `chore(git): do whatever`).

**Comment by ariporad** [comment]
What colors should the labels be for the various commands?

**Comment by nfischer** [comment]
Any color is fine with me. Doesn't matter to me very much.

**Comment by ariporad** [comment]
@nfischer: I'm sorry I haven't had much time to devote to this recently. I've been really busy. I should be more available over the next few days.

P.S. Do you ever sleep? Or do you just work on this project 24/7?

**Comment by arturadib** [comment]
Hey guys– Wow, awesome energy all around! :)

I've created a Gitter room at https://gitter.im/shelljs/shelljs. I've also added both of you as members of the shelljs org, so you should be members there too I think.

As to integrations, our highest priority is CI testing on Windows since we don't have any coverage there, so feel free to set that up!

As for other tools (code style/conventions, semver, etc) why don't you spend some time doing things the old/manual way to get a feel for the problem in this setting and then if that proves to be too inefficient maybe we can think about other integrations?

Lastly, please let me know your npm usernames so I can add you as contributors there too :)

Many thanks for your contributions!!

**Comment by arturadib** [comment]
@nfischer btw you should totally create a Twitter account so I can give you proper credit there! :)

**Comment by ariporad** [comment]
@arturadib: Thanks so much!!!! You rock!!!

I like to be creative with usernames, so I'm @ariporad on npm :smile:.

How involved do you want to be going forward? Do you want to be involved in all decisions, or nothing?

**Comment by ariporad** [comment]
@nfischer: do you want to keep the commit message guidelines? 

And does anyone have any objections to me setting up the following:

1. LGTM.io
2. Gitcop (if we keep the guidelines)
3. AppVeyor
4. ???

Thanks again @arturadib!

**Comment by arturadib** [comment]
@ariporad not a big fan of imposing commit message styles to be honest

AppVeyor sounds great if it lets us do CI on Windows, which we need badly

**Comment by arturadib** [comment]
@ariporad added you on npmjs.com

**Comment by ariporad** [comment]
@arturadib: Thanks so much! We'll hold of on the commit messages for now, they could be very useful in the future.

**Comment by ariporad** [comment]
@arturadib: Sorry, but it looks like you need to change a setting for the github org for us to be able to setup AppVeyor/LGTM/etc. Would you mind doing that? Thanks!

**Comment by nfischer** [comment]
@arturadib
My Twitter: ntfschr
My npm: nfischer
Thanks!

**Comment by nfischer** [comment]
I think CI should be our first priority. Let's add OS X support (I'll update my PR tonight).

After that we can focus on Windows support.

Also, I think it would be good to find a time when we can all get on the gitter and lay down concrete goals and set up roles for the project to get things rolling

**Comment by ariporad** [comment]
@nfischer: Agreed. Although it's kind of bad timing, as I'm going on vacation this on Saturday, and I'm not sure if I'll have internet access. We could maybe do one night this week, otherwise in two weeks when I'm back.

**Comment by ariporad** [comment]
@nfisher, @arturadib: Im on vacation, so I don't have great internet, but I'd like to make a time to get together the three of us in gitter and discuss the future. (Any day in the next week or so works for me).

Im really sorry about being pretty inactive recently, this was really poor timing on my part. (And Ive been trying to at least check in from my phone when I can, so that usually means reviewing PRs, not contributing code, which makes me feel just a little like a taskmaster, but i promise it will change soon).

**Comment by nfischer** [comment]
Sounds good to me. I pinged the gitter the other day to see when your availability would be. I can be free sometime tomorrow and have a fairly light schedule during the week. Would tomorrow work for you?

**Comment by ariporad** [comment]
@nfisher: I'm not sure when "tomorrow" is (based off your comment), but I pinged the gitter. Let's move this there.

**Comment by ariporad** [comment]
I'm also going to lock this.
